plugins/parsers/avro/schema_registry.go

```diff
@@ -14,21 +18,61 @@ type schemaAndCodec struct {
 }
 
 type schemaRegistry struct {
-	url   string
-	cache map[int]*schemaAndCodec
+	url        string
+	authBase64 string
+	cache      map[int]*schemaAndCodec
+	client     http.Client
 }
 
 const schemaByID = "%s/schemas/ids/%d"
 
-func newSchemaRegistry(url string) *schemaRegistry {
-	return &schemaRegistry{url: url, cache: make(map[int]*schemaAndCodec)}
+func newSchemaRegistry(url string, authBase64 string, caCertPath string) (*schemaRegistry, error) {
+	var client *http.Client
+
+	caCert, err := os.ReadFile(caCertPath)
+	if err != nil {
+		return nil, err
+	}
+
+	if caCertPath != "" {
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+		client = &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					RootCAs: caCertPool,
+				},
+				MaxIdleConns:    10,
+				IdleConnTimeout: 90 * time.Second,
+			},
+		}
+	} else {
+		client = &http.Client{
+			Transport: &http.Transport{
+				MaxIdleConns:    10,
+				IdleConnTimeout: 90 * time.Second,
+			},
+		}
+	}
+
+	return &schemaRegistry{url: url, authBase64: authBase64, cache: make(map[int]*schemaAndCodec), client: *client}, nil
 }
 
 func (sr *schemaRegistry) getSchemaAndCodec(id int) (*schemaAndCodec, error) {
 	if v, ok := sr.cache[id]; ok {
 		return v, nil
 	}
-	resp, err := http.Get(fmt.Sprintf(schemaByID, sr.url, id))
+
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf(schemaByID, sr.url, id), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if sr.authBase64 != "" {
+		req.Header.Set("Authorization", fmt.Sprintf("Basic %s", sr.authBase64))
+	}
```

Comment:
After splitting the option into username and password you should do
```suggestion
	if sr.username != "" {
		req.SetBasicAuth(sr.username, sr.password)
	}
```